Added xDnsServerADZone resource and updated README.md

corydwood opened the pull request:
xDnsServerADZone allows you to create AD integrated DNS zones. Updated README.md to include documentation and examples for xDnsServerADZone.

msftclas commented:
Hi __@corydwood__, I'm your friendly neighborhood Microsoft Pull Request Bot (You can call me MSBOT). Thanks for your contribution!
    <span>You've already signed the contribution license agreement. Thanks!</span>
        <p>The agreement was validated by Microsoft and real humans are currently evaluating your PR.</p>

TTYL, MSBOT;

corydwood commented on README.md:
````diff
@@ -66,6 +66,72 @@ Values include: { A-Record | C-Name }
+}
+Sample_xDnsServerReverseADZone
+```
+
````
@TravisEz13 I'm not sure what else you're looking for here. I've included all of the updates to the README for the new resource under the Unreleased section.

iainbrighton replied:
In the ### Unreleased section of the readme, you need to summarise your changes...

corydwood commented:
I'm confused. Below is what I added under the ### Unreleased section. The two bullets at the top are the summary followed by the full documentation and examples. I'm not sure what else you're looking for.


* Added Resource xDnsServerADZone that sets an AD integrated DNS zone.
* Updated README.md with documentation and examples for xDNSServerADZone resource.

#### xDnsServer

The **xDnsServer** DSC resources configure and manage a DNS server. They include **xDnsServerPrimaryZone**, **xDnsServerSecondaryZone**, **xDnsServerADZone**, **xDnsServerZoneTransfer** and **xDnsARecord**.

#### Resources

* **xDnsServerADZone** sets an AD integrated zone on a given DNS server.

##### xDnsServerADZone

* **Name**: Name of the AD DNS zone
* **Ensure**: Whether the AD zone should be present or removed
* **DynamicUpdate**: AD zone dynamic DNS update option.
 * If not specified, defaults to 'Secure'.
 * Valid values include: { None | NonsecureAndSecure | Secure }
* **ReplicationScope**: AD zone replication scope option.
 * Valid values include: { Custom | Domain | Forest | Legacy }
* **DirectoryPartitionName**: Name of the directory partition on which to store the zone.
 * Use this parameter when the ReplicationScope parameter has a value of Custom.
* **ComputerName**: Specifies a DNS server.
 * If you do not specify this parameter, the command runs on the local system.
* **Credential**: Specifies the credential to use to create the AD zone.
 * If you do not specify this parameter, the command runs as the local system.

#### Examples

##### Configuring an AD integrated Forward Lookup Zone

```powershell
configuration Sample_xDnsServerForwardADZone
{
    param
    (
        [pscredential]$Credential,
    )
    Import-DscResource -module xDnsServer
    xDnsServerADZone addForwardADZone
    {
        Name = 'MyDomainName.com'
        DynamicUpdate = 'Secure'
        ReplicationScope = 'Forest'
        ComputerName = 'MyDnsServer.MyDomain.com'
        Credential = $Credential
        Ensure = 'Present'
    }
}
Sample_xDnsServerForwardADZone -Credential (Get-Credential)
```

##### Configuring an AD integrated Reverse Lookup Zone

```powershell
configuration Sample_xDnsServerReverseADZone
{
    Import-DscResource -module xDnsServer
    xDnsServerADZone addReverseADZone
    {
        Name = '1.168.192.in-addr.arpa'
        DynamicUpdate = 'Secure'
        ReplicationScope = 'Forest'
        Ensure = 'Present'
    }
}
Sample_xDnsServerReverseADZone
```

corydwood commented:
@TravisEz13 I've made the requested changes to the README.

I believe the Contributing Guidelines document should be updated to be more clear. Currently it makes it sound like all updates to the README should be placed in the Unreleased section, whereas only a summary of the changes should be placed there and the actual changes should be placed in the correct sections.

iainbrighton commented on DSCResources/MSFT_xDnsServerADZone/MSFT_xDnsServerADZone.psm1:
```diff
@@ -0,0 +1,222 @@
+                Write-Verbose ($LocalizedData.NotDesiredPropertyMessage -f 'DynamicUpdate', $DynamicUpdate, $targetResource.DynamicUpdate)
+                $targetResourceInCompliance = $false
+            }
+            elseif ($targetResource.ReplicationScope -ne $ReplicationScope)
```
Might want to remove the `elseif` and replace with `if`? As it stands, we'll only log one verbose message about a property being incorrect? The Set-TargetResource will still fire.

corydwood replied:
Great suggestion! I just replaced it.

iainbrighton commented:
Looks good to me :+1: Great job!
